mapie/metrics.py

```diff
@@ -307,3 +313,426 @@ def top_label_ece(
+    ...    [False, False, True, True],
+    ...    [True, True, False, True]])
+    >>> print(classification_ssc_score(y_true, y_pred_set, num_bins=2))
+    [0.66666667]
+    """
+    return np.nanmin(classification_ssc(y_true, y_pred_set, num_bins), axis=1)
+
+
+def _gaussian_kernel(
+    x: NDArray,
+    kernel_size: int
+) -> NDArray:
+    """
+    Computes the gaussian kernel of x. (Used in hsic function)
+
+    Parameters
+    ----------
+    x: NDArray
+        The values from which to compute the gaussian kernel.
+    kernel_size: int
+        The variance (sigma), this coefficient controls the width of the curve.
+    """
+    norm_x = x ** 2
+    dist = -2 * np.matmul(x, x.transpose((0, 2, 1))) \
+        + norm_x + norm_x.transpose((0, 2, 1))
+    return np.exp(-dist / kernel_size)
+
+
+def hsic(
+    y_true: NDArray,
+    y_intervals: NDArray,
+    kernel_sizes: ArrayLike = (1, 1)
+) -> NDArray:
+    """
+    Compute the square root of the hsic coefficient. HSIC is Hilbert-Schmidt
+    independence criterion that is a correlation measure. Here we use it as
+    proposed in [4], to compute the correlation between the indicator of
+    coverage and the interval size.
+
+    If hsic is 0, the two variables (the indicator of coverage and the
+    interval size) are independant.
+
+    Warning: This metric should be used only with non constant intervals
+    (intervals of different sizes), with constant intervals the result
+    may be misinterpreted.
+
+    [4] Feldman, S., Bates, S., & Romano, Y. (2021).
+    Improving conditional coverage via orthogonal quantile regression.
+    Advances in Neural Information Processing Systems, 34, 2060-2071.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_intervals: NDArray of shape (n_samples, 2, n_alpha) or (n_samples, 2)
+        Prediction sets given by booleans of labels.
+    kernel_sizes: int
+        The variance (sigma), this coefficient controls the width of the curve.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+        One hsic correlation coefficient by alpha.
+
+    Raises
+    ------
+    ValueError
+        If kernel_sizes has a length different from 2
+        and if it has negative or null values.
+
+    Examples
+    -------
+    >>> from mapie.metrics import hsic
+    >>> import numpy as np
+    >>> y_true = np.array([9.5, 10.5, 12.5])
+    >>> y_intervals = np.array([
+    ... [[9, 9], [10.0, 10.0]],
+    ... [[8.5, 9], [12.5, 12]],
+    ... [[10.5, 10.5], [12.0, 12]]
+    ... ])
+    >>> print(hsic(y_true, y_intervals))
+    [0.31787614 0.2962914 ]
+    """
+    y_true = cast(NDArray, column_or_1d(y_true))
+    y_intervals = check_array_shape_regression(y_true, y_intervals)
+    kernel_sizes = cast(NDArray, column_or_1d(kernel_sizes))
+    if len(kernel_sizes) != 2:
+        raise ValueError(
+            "kernel_sizes should be an ArrayLike of length 2"
+        )
+    if (kernel_sizes[0] <= 0).any():
```

Comment:
You must check for both values

Reply:
Obviously ..

Reply:
and you should add this case to your test, as your tests are passing through this test